src/languages/latex.js

```diff
@@ -1,70 +1,263 @@
 /*
 Language: LaTeX
-Author: Vladimir Moskva <vladmos@gmail.com>
+Author: Benedikt Wilde <bwilde@posteo.de>
 Website: https://www.latex-project.org
 Category: markup
 */
 
+import * as regex from '../lib/regex.js';
+
+/** @type LanguageFn */
 export default function(hljs) {
-  var COMMAND = {
-    className: 'tag',
+  const KNOWN_CONTROL_WORDS = regex.either(...[
+      '(?:NeedsTeXFormat|RequirePackage|GetIdInfo)',
+      'Provides(?:Expl)?(?:Package|Class|File)',
+      '(?:DeclareOption|ProcessOptions)',
+      '(?:documentclass|usepackage|input|include)',
+      'makeat(?:letter|other)',
+      'ExplSyntax(?:On|Off)',
+      '(?:new|renew|provide)?command',
+      '(?:re)newenvironment',
+      '(?:New|Renew|Provide|Declare)(?:Expandable)?DocumentCommand',
+      '(?:New|Renew|Provide|Declare)DocumentEnvironment',
+      '(?:(?:e|g|x)?def|let)',
+      '(?:begin|end)',
+      '(?:part|chapter|(?:sub){0,2}section|(?:sub)?paragraph)',
+      'caption',
+      '(?:label|(?:eq|page|name)?ref|(?:paren|foot|super)?cite)'
+    ].map(word => word + '(?![a-zA-Z@:_])'));
+  const KNOWN_CONTROL_SYMBOLS = /\\|\(|\)|\[|\]|\s|!|,|;/
+  const L3_REGEX = new RegExp([
+      '(?:__)?[a-zA-Z]{2,}_[a-zA-Z](?:_?[a-zA-Z])+:[a-zA-Z]*',
+      '[lgc]__?[a-zA-Z]+_[a-zA-Z](?:_?[a-zA-Z])*[a-zA-Z]',
+      '[qs]__?[a-zA-Z](?:_?[a-zA-Z])+',
+      'use(?:_i)?:[a-zA-Z]*',
+      '(?:else|fi|or):',
+      '(?:if|cs|exp):w',
+      '(?:hbox|vbox):n',
+      '::[a-zA-Z]_unbraced',
+      '::[a-zA-Z:]'
+    ].map(pattern => pattern + '(?![a-zA-Z:_])').join('|'));
+  const L2_VARIANTS = [
+    {begin: /[a-zA-Z@]+/}, // control word
+    {begin: /[^a-zA-Z@]?/} // control symbol
```

Comment:
Could this not use KNOWN_CONTROL_SYMBOLS?  This looks overly broad.

Reply:
Every backslash followed by a non-letter is also a control sequence and many of them are not uncommon. (Just this week I wrote a document where I used `\$` quite a bit.)

Reply:
Then I think we need two variants... one for the more common ones, which can have the default 1 relevance, and one for "this could match almost anything" with no relevance.